src/plugins/intel_cpu/src/nodes/def_conv.cpp

```diff
@@ -1140,10 +1144,21 @@ void DeformableConvolution::DefConvRefExecutor::exec(const float* src, const flo
                         const int v12 = pSampledCoordsVector[sampledCoordIndex + 1];
                         const int v21  = pSampledCoordsVector[sampledCoordIndex + 2];
                         const int v22 = pSampledCoordsVector[sampledCoordIndex + 3];
-                        float val = pInterpWeightsVector[sampledCoordIndex++] * data_im_ptr[v11];  // v11
-                        val += pInterpWeightsVector[sampledCoordIndex++] * data_im_ptr[v12];  // v12
-                        val += pInterpWeightsVector[sampledCoordIndex++] * data_im_ptr[v21];  // v21
-                        val += pInterpWeightsVector[sampledCoordIndex++] * data_im_ptr[v22];  // v22
+
+                        float val = 0;
+                        w11 = pInterpWeightsVector[sampledCoordIndex++];
+                        w12 = pInterpWeightsVector[sampledCoordIndex++];
+                        w21 = pInterpWeightsVector[sampledCoordIndex++];
+                        w22 = pInterpWeightsVector[sampledCoordIndex++];
+
+                        // prevent access to invalid memory:
+                        // 0 * (*wrong_pointer) != 0 in common case, i.e.
```

Comment:
In what scenarion we can get "wrong" pointer?

Reply:
Comment extended. In def.conv., we may get any data as offset, but we must check, if the resulting shift lead out of the memory, so, the scenario is defined by input of offsets.

Reply:
Thanks for clarification.
The thing is I haven't found any guidence in the specification that describes desired behavior in case offsets are out of data tensor bounds. So my questions are:
1, How ngraph reference behaves in this scenarion? I believe we should align with the ngraph reference as most reliable approach.
2. Why you update only CPU reference implementation to handle "wrong" offsets? I expect JIT one should be aligned from the behavior perspective.
3. Why "wrong" offsets comes from the tests? Is it done intentionally or we kust need to fix offsets initialization inside the test?

Reply:
Thank you for making this clear